backend/internal/handlers/classrooms/assignments/assignments.go

```diff
@@ -110,6 +112,122 @@ func (s *AssignmentService) acceptAssignment() fiber.Handler {
 	}
 }
 
+// Generates a token to accept an assignment.
+func (s *AssignmentService) generateAssignmentToken() fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		fmt.Println("Generating assignment token")
+		body := models.AssignmentTokenRequestBody{}
+
+		if err := c.BodyParser(&body); err != nil {
+			fmt.Println("Error parsing body", err)
+			return errs.InvalidRequestBody(body)
+		}
+
+		assignmentID, err := strconv.ParseInt(c.Params("assignment_id"), 10, 64)
+		if err != nil {
+			fmt.Println("Error parsing assignment ID", err)
+			return errs.BadRequest(err)
+		}
+
+		token, err := utils.GenerateToken(16)
+		if err != nil {
+			fmt.Println("Error generating token", err)
+			return errs.InternalServerError()
+		}
+
+		tokenData := models.AssignmentToken{
+			AssignmentID: assignmentID,
+			BaseToken: models.BaseToken{
+				Token:     token,
+				CreatedAt: time.Now(),
+			},
+		}
+
+		// Set ExpiresAt only if Duration is provided
+		if body.Duration != nil {
+			expiresAt := time.Now().Add(time.Duration(*body.Duration) * time.Minute)
+			tokenData.ExpiresAt = &expiresAt
```

Comment:
you actually can't. tokenData.ExpiresAt wants a pointer to a time.Time value, while the result of `time.Now().Add(time.Duration(*body.Duration) * time.Minute)` is a time.Time value. If you tried to put an `&` operator straight in front of the whole line, it will give you an error for trying to take the memory address of a temporary value.